packages/engine/src/scene/components/ModelComponent.ts

```diff
@@ -101,10 +105,19 @@ function ModelReactor({ root }: EntityReactorProps) {
           case '.gltf':
           case '.fbx':
           case '.usdz':
-            const loadedAsset = await AssetLoader.loadAsync(model.src, {
-              ignoreDisposeGeometry: model.generateBVH,
-              uuid
-            })
+            const loadedAsset = await AssetLoader.loadAsync(
+              model.src,
+              {
+                ignoreDisposeGeometry: model.generateBVH,
+                uuid
+              },
+              (onprogress) => {
+                if (hasComponent(entity, SceneAssetPendingTagComponent)) {
+                  setComponent(entity, SceneAssetPendingTagComponent, onprogress.loaded)
+                  getComponent(entity, SceneAssetPendingTagComponent).loadedAmount = onprogress.loaded
```

Comment:
this should read
```ts
setComponent(entity, SceneAssetPendingTagComponent, { loadedAmount: onprogress.loaded })
```

instead of 
```ts
setComponent(entity, SceneAssetPendingTagComponent, onprogress.loaded)
getComponent(entity, SceneAssetPendingTagComponent).loadedAmount = onprogress.loaded
```

Reply:
Ah yes was going to fix this. Weird mistake I made with setcomponent.